ffs/scheduler/internal/sjstore/sjstore.go

```diff
@@ -360,111 +369,169 @@ func (s *Store) GetStartedDeals(iid ffs.APIID, c cid.Cid) ([]cid.Cid, error) {
 	return sd.ProposalCids, nil
 }
 
-// QueuedJobs returns queued jobs for the specified instance id and cids.
-// If the instance id is ffs.EmptyInstanceID, data for all instances is returned.
-// If no cids are provided, data for all data cids is returned.
-func (s *Store) QueuedJobs(iid ffs.APIID, cids ...cid.Cid) []ffs.StorageJob {
+// Select specifies which StorageJobs to list.
+type Select int
+
+const (
+	// All lists all StorageJobs and is the default.
+	All Select = iota
+	// Queued lists queued StorageJobs.
+	Queued
+	// Executing lists executing StorageJobs.
+	Executing
+	// Final lists final StorageJobs.
+	Final
+)
+
+// ListConfig controls the behavior for listing StorageJobs.
+type ListConfig struct {
+	// APIIDFilter filters StorageJobs list to the specified APIID. Defaults to no filter.
+	APIIDFilter ffs.APIID
+	// CidFilter filters StorageJobs list to the specified cid. Defaults to no filter.
+	CidFilter cid.Cid
+	// Limit limits the number of StorageJobs returned. Defaults to no limit.
+	Limit uint64
+	// Ascending returns the StorageJobs ascending by time. Defaults to false, descending.
+	Ascending bool
+	// Select specifies to return StorageJobs in the specified state.
+	Select Select
+	// NextPageToken sets the slug from which to start building the next page of results.
+	NextPageToken string
+}
+
+// List lists StorageJobs according to the provided ListConfig.
+func (s *Store) List(config ListConfig) ([]ffs.StorageJob, bool, string, error) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	var iids []ffs.APIID
-	if iid == ffs.EmptyInstanceID {
-		for iid := range s.queuedJobs {
-			iids = append(iids, iid)
-		}
-	} else {
-		iids = append(iids, iid)
-		ensureJobsSliceMap(s.queuedJobs, iid)
-	}
-
-	var res []ffs.StorageJob
-	for _, iid := range iids {
-		cidsList := cids
-		if len(cidsList) == 0 {
-			for cid := range s.queuedJobs[iid] {
-				cidsList = append(cidsList, cid)
+	byTime := func(ascending bool) func(a, b query.Entry) int {
+		extractTime := func(key string) (int64, error) {
+			parts := strings.Split(key, "/")
+			if len(parts) != 5 {
+				return 0, fmt.Errorf("expected 5 key parts but got %v", len(parts))
 			}
+			return strconv.ParseInt(parts[4], 10, 64)
 		}
-		for _, cid := range cidsList {
-			jobs := s.queuedJobs[iid][cid]
-			for _, job := range jobs {
-				res = append(res, *job)
+		return func(a, b query.Entry) int {
+			l := a
+			r := b
+			if ascending {
+				l = b
+				r = a
+			}
+			lTime, err := extractTime(l.Key)
+			if err != nil {
+				log.Errorf("extracting time from key a: %v", err)
+				return 0
+			}
+			rTime, err := extractTime(r.Key)
+			if err != nil {
+				log.Errorf("extracting time from key b: %v", err)
+				return 0
+			}
+			if lTime > rTime {
+				return -1
+			} else if rTime > lTime {
+				return 1
+			} else {
+				return 0
 			}
```

Comment:
At first, I thought that `if ascending { ` case should be `if !ascending {`, but I see here the conditions are reverted to what I consider the most common understanding of comparisons.

I think in general `left < right` is -1, equal 0, and `left > right` is 1.

Not a point to change code, just a funny comment.